src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/RemoteDiagnosticAnalyzerService.cs

```diff
@@ -2,28 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Diagnostics;
+using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Internal.Log;
 using Microsoft.CodeAnalysis.Remote.Diagnostics;
+using Roslyn.Utilities;
 using RoslynLogger = Microsoft.CodeAnalysis.Internal.Log.Logger;
 
 namespace Microsoft.CodeAnalysis.Remote
 {
-    // root level service for all Roslyn services
-    internal partial class CodeAnalysisService : IRemoteDiagnosticAnalyzerService
+    internal sealed class RemoteDiagnosticAnalyzerService : BrokeredServiceBase, IRemoteDiagnosticAnalyzerService
     {
+        internal sealed class Factory : FactoryBase<IRemoteDiagnosticAnalyzerService>
```

Comment:
so i really don't like the short naming on stuff like this.  i find it actually occludes information.  FOr example, say i'm looking for impls of FactoryBase.  I do "find impls".  It will show a bunch of aline that all show 'sealed class Factory'.  having the type name be more informative is something i find *very* valuable.

Reply:
The factories are dumb types. I think this is much better than having to type
`RemoteDiagnosticAnalyzerService.RemoteDiagnosticAnalyzerServiceFactory`

Reply:
Seems like find all impls can be improved and show the containing type for a nested type like it does for other type members:

![image](https://user-images.githubusercontent.com/41759/92187369-67ffed80-ee0e-11ea-8c32-6528c98d7739.png)


You can also look at the file name:

![image](https://user-images.githubusercontent.com/41759/92187305-3d159980-ee0e-11ea-9771-061f85b17631.png)

Reply:
Filed https://github.com/dotnet/roslyn/issues/47447